SPIKES.md

```diff
@@ -0,0 +1,81 @@
+## Writing a Spike at `conda`
+
+### Summary
+
+In this document, we go over the procedure to follow when create spike issues. Exactly what
+these types of issues are and the steps to take when creating them are all outlined
+below. These guidelines apply for all projects falling underneath the
+[conda](github.com/conda) organization.
+
+### What is a spike?
+
+A spike is a term that is borrowed from extreme programming and agile development.
+They are used when the **outcome of an issue is unknown or even optional**. For example,
+when first coming across a problem that has not been solved before, a team may choose
+to either research the problem or create a prototype in order to better understand it.
+
+Additionally, spikes represent work that **may or may not actually be completed
+or implemented**. An example of this are prototypes created to explore possible solutions.
+Not all prototypes are implemented and the purpose of creating a prototype
+is often to explore the problem space more. For research oriented tasks, the end result
+of this research may be that a feature request simply is not viable at the moment and
+would result in putting a stop to that work.
+
+Finally, spikes are **time boxed**. This means that we set a definite limit on how long
+we want our contributors to work on a spike issue. We do this to prevent contributors
+from falling into a rabbit hole they may never return from (scary!). Instead, we set
+a time limit, typically between 1 and 3 days, to perform work on the spike and then
+have the assignee report back to the development team. If the deliverables defined in
+the spike have not yet been completed, a decision is made on whether it makes sense
+to perform further work on the spike.
+
+Below is what that workflow looks like:
+
+```mermaid
+flowchart LR
+
+A[Create Spike] --> B[Work on spike for\n timebox amount, \n update if necessary]
+B --> C{is it done?}
+C -->|No| E{do we want\nto continue?}
+C -->|Yes| D[Finished]
+E -->|Yes | B
+E -->|No| D
+```
+
+### When do I create a spike?
+
+A spike should be created when we do not have enough information to move forward with solving
+a problem. That simply means whenever we are dealing with unknowns or processes which our
+development team has never encountered before, it may be useful for us to create a spike issue.
+
+In day-to-day work, this may appear when new bug reports or feature requests
+come in which deal with problems or technologies that our development team is unfamiliar with.
+For all issues which our development team has sufficient knowledge of, these issues should
+instead be created as regular issues.
+
+### When do I **not** create a spike?
+
+Below are some common scenarios where creating a spike is not appropriate:
+
+- Writing a technical specification
+- Design work that would go into drafting how a library our API is going to be created
+- Any work that must be completed or is not optional
+
+### How do I create a spike?
+
```

Comment:
```suggestion

Spikes start as a task associated with a regular issue. Only create a spike issue if we determine the spike to be a larger endeavor with multiple deliverables, see the guidelines below.

```

Reply:
Who's "we" here?

Also "see the guidelines below" should be its own sentence.

Reply:
@kenodegard, I have at least one issue with this: spikes don't always start as tasks for a regular issue. For example, creating a prototype wouldn't necessarily always start off with an associated issue. So, let's make sure to accommodate for this as well. 

Perhaps it may be better to first describe what a "mini" spike looks like and then what a spike looks like. Here's how a "mini" spike could look like on a regular issue:

- [ ] Spike description (time box: _1 day_)
  - [ ] Create a report with your finding from the spike; include recommendation if relevant

☝️ As you can see it includes the time box and what we expect as the deliverable. We can assume issue itself would have enough background information.